feat: Add `_meta` field to classes across MCPSchema

pantanurag555 opened the pull request:
## Motivation and Context
`_meta` field was added to multiple classes in the schema to make it easy to pass metadata when communicating between an MCP client and server. Relevant spec PR: https://github.com/modelcontextprotocol/modelcontextprotocol/pull/710

This PR adds `_meta` field across all relevant classes in MCPSchema. It also creates new interfaces for Result and Notifications to ensure that this field is added to any future classes created that implement these interfaces.

Currently MCPSchema in java SDK is missing classes for different list requests (ListResourcesRequest, ListToolsRequest etc-) opting to create the JSONRPCRequest and send it directly to the server instead. Similarly the class for RootsListChangedNotification is missing as well. Therefore, it is not possible to add `_meta` field to these exchanges between the server and client as of now. Relevant classes should be added to the java SDK in the future to maintain parity with the specification. This PR does not address this issue.

Resolves #344 

## How Has This Been Tested?
Added relevant unit tests for all classes.

## Breaking Changes
None. Added relevant backward compatible constructors to ensure

## Types of changes
- [ ] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Documentation update

## Checklist
- [X] I have read the [MCP Documentation](https://modelcontextprotocol.io)
- [X] My code follows the repository's style guidelines
- [X] New and existing tests pass locally
- [X] I have added appropriate error handling
- [X] I have added or updated documentation as needed

tzolov commented on mcp/src/main/java/io/modelcontextprotocol/spec/McpSchema.java:
```diff
@@ -530,80 +561,87 @@ public interface ResourceContent {
+		@JsonProperty("mimeType") String mimeType,
+		@JsonProperty("size") Long size,
+		@JsonProperty("annotations") Annotations annotations,
+		@JsonProperty("_meta") Map<String, Object> meta) implements Annotated {
```
Why the Resource is not a ResourceContent anymore?

pantanurag555 commented:
> @pantanurag555 , thank you for the PR, I've started reviewing it and put some comments. Also making the Resource not implement a ResourceContent is suspicious?
> 
> But since you've made a lot unnecessary formatting changes it is very hard to review. Can you please revert all non-essential changes (like formatting) to make it a more readable.
> 
> cheers

Addressed the feedback. It should be easier to understand when taking a look at the diff. The PR needs to be rebased due to other changes on the same files (which can be quite laborious). Waiting for any additional feedback before attempting the final rebase.

tzolov commented:
Thank you @pantanurag555
